src/makeCalculator.js

```diff
@@ -37,7 +37,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(value) {
+      this.result += value;
+    },
+
+    subtract(value) {
+      this.result -= value;
+    },
+
+    multiply(value) {
+      this.result *= value;
+    },
+
+    divide(value) {
+      this.result /= value;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, value) {
+      callback.call(this, value);
```

Comment:
as I understand call() -  It can be used to invoke (call) a method with an owner object as an argument (parameter).
and in this case you transfer whole object, not a single parametr, not sure is this a god practice here.

With call(), an object can use a method belonging to another object.
Like in case with call we can create even a side object, and take operate method from there. 

will appreciate if you explicate why you use it here.

Reply:
It means that I invoke my `callback` function in context of my object, because in this case I loose my context, and without binding my `this` = window global object. If I invoke `callback` without binding: 

```js
callback(value); // for example I invoke add method

add(value) {
  this.result += value; // without binding it looks like window.result += value
}
// I lost my context now it's window global object not my object
```